src/pymanopt/manifolds/positive_definite.py

```diff
@@ -113,20 +91,221 @@ def exp(self, point, tangent_vector):
+            return point[0]
+        return point
+
+    def random_tangent_vector(self, point):
+        k = self._k
+        n = self._n
+        if k == 1:
+            tangent_vector = multiherm(
+                np.random.randn(n, n) + 1j * np.random.randn(n, n)
+            )
+        else:
+            tangent_vector = multiherm(
+                np.random.randn(k, n, n) + 1j * np.random.randn(k, n, n)
+            )
+        return tangent_vector / self.norm(point, tangent_vector)
+
+    def zero_vector(self, point):
+        k = self._k
+        n = self._n
+        if k == 1:
+            return np.zeros((n, n), dtype=complex)
+        return np.zeros((k, n, n), dtype=complex)
+
+
+class SpecialHermitianPositiveDefinite(HermitianPositiveDefinite):
+    """Manifold of hermitian positive definite matrices with unit determinant.
+
+    Points on the manifold and tangent vectors are represented as arrays of
+    shape ``k x n x n`` if ``k > 1``, and ``n x n`` if ``k == 1``.
+
+    Args:
+        n: The size of matrices in the manifold, i.e., the number of rows and
+            columns of each element.
+        k: The number of elements in the product geometry.
+    """
+
+    def __init__(self, n: int, *, k: int = 1):
+        super().__init__(n, k=k)
+        self.dim = int(k * n * (n + 1) - k)
+
+    def random_point(self):
+        n = self._n
+        k = self._k
+
+        # Generate point on the HPD manifold.
+        point = super().random_point()
+
+        # Unit determinant.
+        shape = (k, 1, 1) if k > 1 else (1, 1)
+        det = (np.linalg.det(point) ** (1 / n)).reshape(shape)
+        point = point / det
+
+        return point
+
+    def random_tangent_vector(self, point):
+        tangent_vector = super().random_tangent_vector(point)
+
+        # Project them on tangent space.
+        tangent_vector = self.projection(point, tangent_vector)
+
+        # Unit norm.
+        tangent_vector = tangent_vector / self.norm(point, tangent_vector)
+
+        return tangent_vector
+
+    def projection(self, point, vector):
+        n = self._n
+        k = self._k
+
+        # Project matrix on tangent space of HPD.
+        vector = super().projection(point, vector)
+
+        # Project on tangent space of SHPD at x.
+        shape = (k, 1, 1) if k > 1 else (1, 1)
+        t = np.real(
+            np.trace(np.linalg.solve(point, vector), axis1=-2, axis2=-1)
+        )
+        t = t.reshape(shape)
+        tangent_vector = vector - (1 / n) * t * point
+
+        return tangent_vector
+
+    def euclidean_to_riemannian_gradient(self, point, euclidean_gradient):
+        return self.projection(
+            point,
+            super().euclidean_to_riemannian_gradient(
+                point, euclidean_gradient
+            ),
+        )
+
+    @_raise_not_implemented_error
+    def euclidean_to_riemannian_hessian(
+        self, point, euclidean_gradient, euclidean_hessian, tangent_vector
+    ):
+        pass
```

Comment:
Functions prefixed with underscore should be treated as private to the module they're defined it. So we can either
1. Rename the decorator to indicate it's fine to import the function in other modules,
2. Replace the snippet above with this which automatically raises the desired exception but looks a bit mysterious:
```suggestion
    euclidean_to_riemannian_hessian = Manifold.euclidean_to_riemannian_hessian
```

Reply:
I prefer the first option which is more explicit. I made the changes accordingly.